src/content/guides/asset-modules.md

```diff
@@ -0,0 +1,353 @@
++       type: 'asset/resource'
++     }
++   ]
++ }
+};
+```
+
+__src/index.js__
+
+``` js
+import mainImage from './images/main.png';
+
+// ...
+
+img.src = mainImage; // '/dist/151cfcfa1bd74779aadb.png'
+```
+
+All `png` files will be emitted to the output directory and their paths will be injected into the bundles. Just like with `file-loader`
+
+### Custom output filename
+
+By default, `asset/resource` modules are emitting with `[hash][ext]` filename into output directory.
+
+You can modify this template with `output.assetModuleFilename` in your config:
+
+__webpack.config.js__
+
+``` diff
+const path = require('path');
+
+module.exports = {
+  entry: './src/index.js',
+  output: {
+    filename: 'main.js',
+    path: path.resolve(__dirname, 'dist'),
++   assetModuleFilename: 'images/[hash][ext]'
+  },
+  experiments: {
+    asset: true
+  },
+  module: {
+    rules: [
+      {
+        test: /\.png/,
+        type: 'asset/resource'
+      }
+    ]
+  }
+};
+```
+
+## Inlined assets
+
+__webpack.config.js__
+
+``` diff
+const path = require('path');
+
+module.exports = {
+  entry: './src/index.js',
+  output: {
+    filename: 'main.js',
+    path: path.resolve(__dirname, 'dist')
+  },
+  experiments: {
+    asset: true
+  },
+  module: {
+    rules: [
+      {
+-       test: /\.png/,
+-       type: 'asset/resource'
++       test: /\.svg/,
++       type: 'asset/inline'
+      }
+    ]
+  }
+};
+```
+
+__src/index.js__
+
+``` diff
+- import mainImage from './images/main.png';
++ import metroMap from './images/matro.svg';
+
+// ...
+
+- img.src = mainImage; // '/dist/151cfcfa1bd74779aadb.png'
++ block.style.background = `url(${metroMap}); // url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDo...vc3ZnPgo=)
+```
+
+All `svg` files will be injected into the bundles as data-url. Just like with `url-loader`
+
+### Custom data-url generator
+
+By default, data-url is a file content encoded with base64.
+
+If you want to use a custom encoding algorithm, you may specify a custom function to encode a file content:
+
+__webpack.config.js__
+
+``` diff
+const path = require('path');
++ const svgToMiniDataURI = require('mini-svg-data-uri');
+
+module.exports = {
+  entry: './src/index.js',
+  output: {
+    filename: 'main.js',
+    path: path.resolve(__dirname, 'dist')
+  },
+  experiments: {
+    asset: true
+  },
+  module: {
+    rules: [
+      {
+        test: /\.svg/,
+        type: 'asset/inline',
++       generator: {
++         dataUrl: content => {
++           content = content.toString();
++           return svgToMiniDataURI(content);
++         }
++       }
+      }
+    ]
+  }
+};
+```
+
+This one will use `mini-svg-data-uri` package to encode all `svg` files.
+
+## Source assets
+
+__webpack.config.js__
+
+``` diff
+const path = require('path');
+- const svgToMiniDataURI = require('mini-svg-data-uri');
+
+module.exports = {
+  entry: './src/index.js',
+  output: {
+    filename: 'main.js',
+    path: path.resolve(__dirname, 'dist')
```

Comment:
assetModuleFilename: 'images/[hash][ext]' is missing in this snippet or it should be removed in a diff in one of the above snippets